documentation/course/e2/README.md

```diff
@@ -2069,15 +2074,55 @@ TODO @jakubzehner: Dodać diagram pakietów, opis architektury i endpointy.
+
+### API
+
+#### API publiczne
+
+| **Rola**                                    | **Metoda** | **Endpoint**              | **Wymagania**                          | **Opis**                                       |
+| ------------------------------------------- | ---------- | ------------------------- | -------------------------------------- | ---------------------------------------------- |
+| `passenger`, `admin`                        | `GET`      | `/ext/v1/lines`           | `LOG/01`, `LOG/02`                     | Pobranie listy aktywnych linii.                |
+| `passenger`, `admin`                        | `GET`      | `/ext/v1/lines/:id`       | `LOG/01`, `LOG/02`                     | Pobranie informacji o linii.                   |
+| `admin`                                     | `POST`     | `/ext/v1/lines`           | `LOG/03`                               | Utworzenie nowej linii.                        |
+| `admin`                                     | `PATCH`    | `/ext/v1/lines/:id`       | `LOG/04`                               | Edytowanie linii.                              |
+| `admin`                                     | `DELETE`   | `/ext/v1/lines/:id`       | `LOG/05`                               | Dezaktywacja linii.                            |
+| `passenger`, `admin`                        | `GET`      | `/ext/v1/lines/:id/stops` | `LOG/06`, `LOG/07`                     | Pobranie listy przystanków konkretnej linii.   |
+| `passenger`, `admin`                        | `GET`      | `/ext/v1/stops/:id/lines` | `LOG/06`, `LOG/07`                     | Pobranie listy linii obsługujących przystanek. |
+| `admin`                                     | `POST`     | `/ext/v1/stops/`          | `LOG/08`                               | Utworzenie nowego przystanku.                  |
+| `admin`                                     | `PATCH`    | `/ext/v1/stops/:id`       | `LOG/09`                               | Edytowanie przystanku.                         |
+| `admin`                                     | `DELETE`   | `/ext/v1/stops/:id`       | `LOG/10`                               | Dezaktywacja przystanku.                       |
+| `passenger`, `admin`                        | `GET`      | `/ext/v1/routes`          | `LOG/11`, `LOG/12`                     | Pobranie rozkładu jazdy.                       |
```

Comment:
**question:** When would this endpoint be called? I don't see a business need to get a list of ALL routes in any app at any time. Does it support filtering?